pkg/service/backup_handler.go

```diff
@@ -101,6 +127,48 @@ func (h *BackupHandler) runFullBackupInternal(now time.Time) error {
 	return nil
 }
 
+func (h *BackupHandler) startFullBackupForAllNamespaces(upperBound time.Time, client *aerospike.Client) error {
+	clear(h.fullBackupHandlers)
+
+	options := shared.BackupOptions{}
+	if h.backupFullPolicy.IsSealed() {
+		options.ModBefore = &upperBound
+	}
+
+	for _, namespace := range h.namespaces {
+		backupFolder := getFullPath(h.backend.fullBackupsPath, h.backupFullPolicy, namespace, upperBound)
+		backupPath := h.backend.wrapWithPrefix(backupFolder)
+		handler, err := backupService.BackupRun(h.backupRoutine, h.backupFullPolicy, client,
+			h.storage, h.secretAgent, options, &namespace, backupPath)
+		if err != nil {
+			backupFailureCounter.Inc()
+			return fmt.Errorf("could not start backup of namespace %s, routine %s: %w", namespace, h.routineName, err)
+		}
+
+		h.fullBackupHandlers[namespace] = handler
+	}
+
+	return nil
+}
+
+func (h *BackupHandler) waitForFullBackups(ctx context.Context, now time.Time) error {
+	startTime := time.Now()
```

Comment:
How does this `startTime` differ from the `now` parameter?

Reply:
startTime is used only to calculate duration.
now — is a execution time from scheduler, we use it in business logic.
They might be almost equal, but still the meaning is different.